lib/senkyoshi/models/rule.rb

```diff
@@ -0,0 +1,45 @@
+require "senkyoshi/models/grade_completed_criteria"
+require "senkyoshi/models/content_reviewed_criteria"
+require "senkyoshi/models/grade_range_criteria"
+require "senkyoshi/models/grade_range_percent_criteria"
+require "senkyoshi/models/resource"
+
+module Senkyoshi
+  class Rule < FileResource
+    attr_reader(:title, :content_id, :criteria_list, :id)
+
+    CRITERIA_MAP = {
+      grade_completed_criteria: GradeCompletedCriteria,
+      content_reviewed_criteria: ContentReviewedCriteria,
+      grade_range_criteria: GradeRangeCriteria,
+      grade_range_percent_criteria: GradeRangePercentCriteria,
+    }.freeze
+
+    def initialize(resource_id)
+      super(resource_id)
+      @criteria_list = []
+    end
+
+    def get_criteria_list(xml)
+      xml.children.select { |child_xml| !child_xml.blank? }.
+        map do |child_xml|
+          CRITERIA_MAP[child_xml.name.downcase.to_sym]&.from_xml(child_xml)
```

Comment:
This is awesome Ruby 2.3 syntax, but we need to support down to ruby 2.1.  Please fix to accommodate.